iocore/hostdb/HostDB.cc

```diff
@@ -1588,6 +1588,12 @@ HostDBContinuation::set_check_pending_dns()
   this->setThreadAffinity(this_ethread());
   HostDBContinuation *c = q.head;
   for (; c; c = static_cast<HostDBContinuation *>(c->link.next)) {
+    if (c == this) {
+      Warning("Skip the insertion of the same continuation to pending dns");
+      return false;
+    }
+  }
+  for (c = q.head; c; c = static_cast<HostDBContinuation *>(c->link.next)) {
     if (hash.hash == c->hash.hash) {
```

Comment:
The `hash.hash == c->hash.hash` must be `true` if `c` is `this`.

Reply:
@oknet you suggested using in_or_enqueue(), however, if the continuation is not in the list, we need to check the hash matched or not. I guess the latest version of using in() works well.